test/regexp/bytes/nonUTFIO.chpl

```diff
@@ -0,0 +1,127 @@
+/* stdin of this test contains:
+   4 (almost) identical lines like "Pattern:Capture" but they have different
+   UTF8 characteristics
+
+   Pattern:Capture
+   Patt\xffern:Capture
+   Patt\xffern:Cap\xffture
+   Pattern:Cap\xffture
+*/
+
+use IO;
+
+{
+  var f = openmem();
+
+  var w = f.writer();
+  w.write("Pattern:Capture\n");
+  w.write(b"Patt\xffern:Capture\n");
+  w.write(b"Patt\xffern:Cap\xffture\n");
+  w.write(b"Pattern:Cap\xffture\n");
+  w.close();
+
+  // read from it -- following four reads are how these should be read
+  var r = f.reader();
+  var captureString: string;
+  var captureBytes: bytes;
```

Comment:
Can you add some testing with passing a compiled bytes/string regexp to this test?

Reply:
Done. Current version of this test does that via the testRead helper.